citesphere/src/main/java/edu/asu/diging/citesphere/web/user/authorities/AuthorityEntryController.java

```diff
@@ -71,8 +84,71 @@
             logger.info("Not a valid URI: " + uri, e);
             return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
         }
-     
-        entry = authorityService.create(entry, (IUser)authentication.getPrincipal());
+
+        entry = authorityService.create(entry, (IUser) authentication.getPrincipal());
+        return new ResponseEntity<IAuthorityEntry>(entry, HttpStatus.OK);
+    }
+    
+    @RequestMapping(value = "/auth/authority/conceptpower/create", method = RequestMethod.POST)
+    public ResponseEntity<IAuthorityEntry> createConceptpowerAuthorityEntry(Authentication authentication,
+            @RequestParam("uri") String uri) {
+        IAuthorityEntry entry = null;
+        for(IAuthorityEntry authorityEntry : authorityConceptpowerResult.getFoundAuthorities()) {
+            if(authorityEntry.getUri().equals(uri)) {
+                entry = authorityService.create(authorityEntry, (IUser) authentication.getPrincipal());
+                break;
+            }
+        } 
         return new ResponseEntity<IAuthorityEntry>(entry, HttpStatus.OK);
     }
+
+    @RequestMapping("/auth/authority/{zoteroGroupId}/find/userAuthorities")
+    public ResponseEntity<AuthoritySearchResult> getUserAuthorities(Model model, Authentication authentication,
+            @PathVariable("zoteroGroupId") String zoteroGroupId,
+            @RequestParam(defaultValue = "0", required = false, value = "page") int page,
+            @RequestParam(defaultValue = "10", required = false, value = "pageSize") int pageSize,
+            @RequestParam("firstName") String firstName, @RequestParam("lastName") String lastName) {
+
+        AuthoritySearchResult authorityResult = new AuthoritySearchResult();
+
+        List<IAuthorityEntry> userEntries = authorityService.findByName((IUser) authentication.getPrincipal(),
+                firstName, lastName, page, pageSize);
+
+        authorityResult.setFoundAuthorities(userEntries);
+        authorityResult.setCurrentPage(page + 1);
+        authorityResult.setTotalPages(authorityService
+                .getTotalUserAuthoritiesPages((IUser) authentication.getPrincipal(), firstName, lastName, pageSize));
+        return new ResponseEntity<AuthoritySearchResult>(authorityResult, HttpStatus.OK);
+    }
+    
+    @RequestMapping("/auth/authority/{zoteroGroupId}/find/searchAuthorities/{source}")
+    public ResponseEntity<AuthoritySearchResult> getConceptpowerAuthorities(Authentication authentication,
```

Comment:
this method does not just search conceptpower, does it? It could be used for any of the authority services (e.g. viaf). This means that you would override the results of one authority service with the results of the next one.

Reply:
Also, let's move this method up so the methods are more or less in the order in which they are called from the frontend,

Reply:
Are you suggesting to make it a generic method and update the logic so that one authority results will not override others or you are suggesting to update the URL and make it specific to conceptpower?

Reply:
yes, to the first. It should be generic and not override the results of the other searches.